Automattic/wp-calypso
# Framework: Drop redux form

Historically, we've had several instances where we used `redux-form` for handling forms in Calypso, however in recent efforts we've either been removing or refactoring them. 

At the time of creating this PR, there is a single instance that uses them - the Zoninator extension. 

This PR refactors that usage and removes `redux-form` altogether from Calypso.

#### Changes proposed in this Pull Request

* Extensions: Refactor Zoninator forms to not use `redux-form` and fall back to a Redux implementation.
* Components: Remove unused redux-forms components
* State: Remove unused form state
* Test: Remove `redux-form` from jest `transformIgnorePatterns`
* Remove `redux-form` dependency

#### Testing instructions

* Go to `/extensions/zoninator/:site` where `:site` is a Jetpack site with the Zoninator plugin installed and activated.
* Test the entire Zoninator interface in Calypso (not necessarily in that order) and verify it's still working correctly, including:
  * Add a new Zone
  * Edit an existing Zone's title and description
  * Add zone content to a zone
  * Edit existing zone content - adding new posts, removing existing ones, reordering.
* Verify all tests pass.
